crates/core/tedge_api/tests/fixtures/invalid/reject_measurement_named_type.expected_error

```diff
@@ -1,4 +1,4 @@
-Invalid JSON: Invalid measurement name: "type" is a reserved word. at line 3 column 8: `": 40,
+Invalid JSON: invalid type: integer `40`, expected a borrowed string at line 3 column 12: `0,
```

Comment:
I think that with this change, thin-edge json should accept a "type" value being a number or a string.

- In the first case, the int value will be seen as a measurement by thin-edge json, but seen as an error when translated to c8y.
- In the second case, the string value will be seen as an unknown fragment by thin-edge json, and interpreted as a measurement type by the c8y mapper.